### ISSUE TITLE
Is it possible to extend this to support dragging between lists?

### ISSUE THREAD

**Issue by mattgi** [rodrigodagostino/svelte-sortable-list#2]
Very clean library and eager to use it for my use case.

However, I am wondering if its possible to support multiple lists where I can drag items between lists, but also reorder the lists themselves.

So a structure like:
```
<script lang="ts">
  import { SortableList, sortItems } from "@/components/behavior/sortable";
  let items = [
    {
      id: 1,
      children: [
        { id: 1.1, children: [] },
        { id: 1.2, children: [] },
      ],
    },
    {
      id: 2,
      children: [
        { id: 2.1, children: [] },
        { id: 2.2, children: [] },
        { id: 2.3, children: [] },
      ],
    },
    {
      id: 3,
      children: [
        { id: 3.1, children: [] },
        { id: 3.2, children: [] },
        { id: 3.3, children: [] },
        { id: 3.4, children: [] },
      ],
    },
  ];

  function handleSort(event: CustomEvent) {
    const { prevIndex, nextIndex } = event.detail;
    items = sortItems(items, prevIndex, nextIndex);
  }
</script>

<SortableList {items} let:item direction="horizontal" on:sort={handleSort}>
  <div class="border p-4 justify-center items-center flex flex-col">
    {item.id}
    <SortableList items={item.children} let:item direction="vertical">
      <div class="border p-4 justify-center items-center flex flex-col">
        {item.id}
      </div>
    </SortableList>
  </div>
</SortableList>
```

Ideally, using `<svelte:self />` it could be nested deeply. Do you think this is achievable?

**Comment by rodrigodagostino** [comment]
Hi, @mattgi! :) My apologies for the delay, it’s been a busy week.

Well, I’ve got some good news for you, because I’m currently working on an important refactoring to improve the overall flexibility of the package, and I’m hoping it will help facilitate something like nested lists. I’m don’t think it will be possible to drag and drop items between lists out of the box, but nesting lists should be feasible.

It’s hard for me to give you an expected delivery date, but I’ll drop another comment when this is available ;)

**Comment by rodrigodagostino** [comment]
Hey, @mattgi! That took a long while, but I’m back with some good news :)

I’ve finished the first and main stage of the major refactoring, which I just published an hour ago. It is now possible to create the kind of structure you were looking for, since I’ve extracted the `<SortableItem>` out of the `<SortableList>` component, allowing them to be imported separately.

I did a quick attempt to replicate your example, and even though the lists are displayed correctly, that’s about it. There are several bugs all around preventing it from working as desired. But I got really curious about it, so I will probably be working on it in the next few days to allow for proper nested lists :)

If you have the time, please, give it a try and let me know your thoughts.

**Comment by ManuelPuentes** [comment]
Hey! @rodrigodagostino  i was trying to update my code with your new realease but i ve been facing an issue, my sortableItemData looks like this  
``` 	
let items: SortableItemData[] = [
	{
	  id: "list-item-1",
	  text: "List item 1",
	  files: ["file1.svg", "file2.svg", "file3.svg", "file4.svg"],
  }];
```

when i tied inside the SortableItem component to iterate through files array to create 1 component per file im getting this 
```
Argument of type 'unknown' is not assignable to parameter of type 'ArrayLike<unknown> | Iterable<unknown>'.

```
i have a nested list but not a sortable a normal list, can you helped me pls, older code is this

```
 "@rodrigodagostino/svelte-sortable-list": 0.9.6",

	<SortableList {items} let:item on:sort={handleSort}>
		<SortIcon
			id="handle-icon"
			slot="handle"
		/>

		<Tooltip triggeredBy="#handle-icon" placement="left">Drag!</Tooltip>

		<Folder folder_name={item.trait}>
			{#each item.files as file}
				<File file_name={file} folder_name={item.trait} />
			{/each}
		</Folder>
	</SortableList>
```

**Comment by rodrigodagostino** [comment]
Hi, @ManuelPuentes! Thanks for reaching out :)

I’m a little confused. I seems that you are using version `v0.9.6`, but the type definition for `SortableItemData` was not available back then. I reproduced your scenario with the latest available version (`0.10.1`), and even tried it in one of my own apps, but I’m not seeing any kind of errors.

If you’re working with version `0.9.6`, try replacing `SortableItemData[]` with `SortableListProps['items']` (check the [README for v0.9.6](https://github.com/rodrigodagostino/svelte-sortable-list/tree/v0.9.6)).

If you need further assistance, I would appreciate it if you could open up a new issue. Since this comments are not related to the current issue, we’re derailing the conversation here :P

**Comment by ManuelPuentes** [comment]
> Hi, @ManuelPuentes! Thanks for reaching out :)
> 
> I’m a little confused. I seems that you are using version `v0.9.6`, but the type definition for `SortableItemData` was not available back then. I reproduced your scenario with the latest available version (`0.10.1`), and even tried it in one of my own apps, but I’m not seeing any kind of errors.
> 
> If you’re working with version `0.9.6`, try replacing `SortableItemData[]` with `SortableListProps['items']` (check the [README for v0.9.6](https://github.com/rodrigodagostino/svelte-sortable-list/tree/v0.9.6)).
> 
> If you need further assistance, I would appreciate it if you could open up a new issue. Since this comments are not related to the current issue, we’re derailing the conversation here :P

no, i missexplained myself sorry, the older code version was using 0,9.6 but then new one is using 10.0, the problem mentioned above is for 10.0. the code  shared was the older version and  my question was how can i adappted to 10.0 ? my solution was to avoid using "SortableItemData" type on items array

**Comment by rodrigodagostino** [comment]
Hey, @mattgi! just wanted to let you know that, due to lack of activity, I will be closing this issue. But feel free to open it back again in case there is something that we could still work on :)
